PR2. Add annotation manifest for pre-computed per-tool inputs (#90)
EBI-Metagenomics/mobilome-annotation-pipeline: docs/annotation_manifest.md

Reviewer: Should we just have one samplesheet with all the columns instead?

Author: That's what I was trying to avoid. This pipeline is quite popular among external users, so I wanted to keep the basic usage clean

# Annotation Manifest

> The `--annotation_manifest` option allows functional annotation outputs produced by the [EBI Genomes Catalogues pipeline](https://github.com/EBI-Metagenomics/genomes-catalogue-pipeline) to be reused directly, bypassing the corresponding internal tools. This avoids redundant compute when processing genomes that have already been annotated in a catalogue context.

Reviewer: ```suggestion
The `--annotation_manifest` option allows functional annotation outputs produced by the [MGnify genomes catalogue pipeline](https://github.com/EBI-Metagenomics/genomes-catalogue-pipeline) to be reused directly, bypassing the corresponding internal tools. This avoids redundant compute when processing genomes that have already been annotated in a catalogue context.
```

## When to use it

Use `--annotation_manifest` when:

> - You are running MAP on assemblies that are part of an EBI Genomes Catalogue run and you already have InterProScan, AMRFinderPlus, antiSMASH, GECCO, or SanntiS outputs for those assemblies.

Reviewer: ```suggestion
- You are running MAP on assemblies that are part of an MGnify genomes catalogue pipeline run and you already have InterProScan, AMRFinderPlus, antiSMASH, GECCO, or SanntiS outputs for those assemblies.
```

- You want to skip re-running one or more annotation tools to reduce runtime or avoid database requirements.

> When a manifest is provided, the pipeline skips the corresponding internal runs (e.g. does not invoke InterProScan or AMRFinderPlus internally) and instead routes the manifest files directly into the downstream integration and reporting steps. Functional annotation tools not covered by the manifest still run normally.

Reviewer: ```suggestion
When a manifest is provided, the pipeline skips the corresponding internal runs (e.g. does not invoke InterProScan or AMRFinderPlus) and instead routes the manifest files directly into the downstream integration and reporting steps. Functional annotation tools not covered by the manifest still run normally.
```

MGYG000012345,/data/catalogues/MGYG000012345_InterProScan.tsv,,,, 
```

In the second row, only the IPS TSV is provided; all BGC tools will run internally for that sample.

## Column details

| Column | Source tool | Used by |
|---|---|---|
| `ips_tsv` | InterProScan | PATHOFACT2 (CDS search step), COMBINEREPORTER |
| `amrfinder_tsv` | AMRFinderPlus (catalogue layout) | AMR_ANNOTATION subworkflow (REFORMAT step), COMBINEREPORTER |
| `antismash_gff` | antiSMASH | BGC_ANNOTATION subworkflow, COMBINEREPORTER |
| `gecco_gff` | GECCO | BGC_ANNOTATION subworkflow, COMBINEREPORTER |
| `sanntis_gff` | SanntiS | BGC_ANNOTATION subworkflow, COMBINEREPORTER |

### AMRFinderPlus column layout

The AMRFinderPlus output produced by the genomes-catalogue-pipeline uses a column layout that differs from the MAP-internal run. MAP automatically normalises the catalogue-format TSV via an internal reformat step before integration, so no manual preprocessing is required.

## Interaction with skip flags

Manifest columns and skip flags are independent:

> - If a column is populated in the manifest, the corresponding internal tool is bypassed regardless of skip flags.

Reviewer: ```suggestion
- If a column is populated in the manifest, the corresponding tool is bypassed regardless of skip flags.
```

- If a column is empty and the skip flag is set (e.g. `--skip_amrfinderplus`), the tool is skipped and no results for that tool appear in the report.

> - If a column is empty and the skip flag is not set, the tool runs internally as usual.

Reviewer: ```suggestion
- If a column is empty and the skip flag is not set, the tool runs as usual.
```

## Invocation

```bash

> nextflow run main.nf \

Reviewer: ```suggestion
nextflow run ebi-metagenomics/mobilome-annotation-pipeline \
```

    --input samplesheet.csv \
    --annotation_manifest manifest.csv \
    -c my_paths.config \
    -profile singularity
```

> The manifest is validated against `assets/schema_manifest.json` at startup. Sample IDs that appear in the manifest but not in the samplesheet are ignored with a warning.

Reviewer: ... I'm inclined to make it fail instead. We could miss analyses because of a silly error in a csv.

Author: I agree. I will make the change.

Author: I have added a test as well at subwf level. I need to make more changes in the workflow on PR4 and add the exception to e2e test there
